cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -837,7 +840,9 @@ public MapAndStatus(T loadMap, K status) {
                 return requestRebalance(reconciliation, host, apiClient, kafkaRebalance, true, rebalanceOptionsBuilder);
             default:
                 LOGGER.warnCr(reconciliation, "Ignore annotation {}={}", ANNO_STRIMZI_IO_REBALANCE, rebalanceAnnotation);
-                return Future.succeededFuture(new MapAndStatus<>(null, buildRebalanceStatusFromPreviousStatus(kafkaRebalance.getStatus(), validate(reconciliation, kafkaRebalance))));
+                Set<Condition> conditions = validate(reconciliation, kafkaRebalance);
+                conditions.add(StatusUtils.buildWarningCondition("InvalidAnnotation", "Unsupported/Unknown annotation value: " + rawRebalanceAnnotation(kafkaRebalance) + ". The rebalance resource supports the following annotations : approve, refresh, stop."));
```

Comment:
Don't use space before `:`:

```suggestion
                conditions.add(StatusUtils.buildWarningCondition("InvalidAnnotation", "Unsupported/Unknown annotation value: " + rawRebalanceAnnotation(kafkaRebalance) + ". The rebalance resource supports the following annotations: approve, refresh, stop."));
```

Reply:
It looks like we set this condition only when the proposal is Ready. Is that intentional? When happens when a wrong annotation is used when the proposal is in a different state?

Reply:
@scholzj The annotations that cruise control support are generally applied in `ProposalReady` state mainly. As per the code I think if we are on `Ready` or maybe `Stop` state, we can only use `refresh` annotation and other annotations are ignored in that case basically. So I guess in that case, even other two correct annotations are wrong so I can add a warning regarding that ?

Reply:
I think you need to consult the CC experts what is the best approach. This is just something what seemed strange to me.

Reply:
@ShubhamRwt IIRC we agreed offline to set a warning condition in any case a not supported annotation was used on the `KafkaRebalance` any state it is. So for example, in `Stop` state we are accepting `refresh` only but we are ignoring the others logging the warning. I would say that we should add a warning condition in that case as well. The same for `Ready`state.
The rule should be: in any state we are if an annotation is not supported, we should add a warning condition and not just logging the warning as we are doing today.

Reply:
@ppatierno I have added the warning in other states in the latest commit

Reply:
Ok so I guess the @scholzj comment was stale and you made the change, I see it but ...
What I see in other states (i.e. onNotReady) is that a local `conditions` is created but never used in the sense that it's not reflected in a built status for the KafkaRebalance resource.

```java
} else {
            // Stay in the current NotReady state, returning null as next state
            Set<Condition> conditions = validate(reconciliation, kafkaRebalance);
            conditions.add(StatusUtils.buildWarningCondition("InvalidAnnotation", "Wrong annotation value: " + rawRebalanceAnnotation(kafkaRebalance) + ". The rebalance resource supports the following annotations in `NotReady` state: refresh"));
            return Future.succeededFuture(new MapAndStatus<>(null, null));
        }
```

I guess you would need to build the status as well using the warning conditions.

Reply:
@ppatierno @scholzj I have added the warning test now with some change in how I was writing the test. They seem to work fine now and applied on all states